client_test.go

```diff
@@ -135,6 +135,53 @@ func TestAPIRequest(t *testing.T) {
 	}
 }
 
+func TestBuildApiRequestFailure(t *testing.T) {
+	tests := []struct {
+		name              string
+		expectedError     error
+		expectedErrorType interface{}
+		path              string
+		clusterDown       bool
+	}{
+		{
+			name:          "cluster down",
+			expectedError: ErrMarathonDown,
+			clusterDown:   true,
+		},
+		{
+			name:              "invalid request parameter",
+			expectedErrorType: newRequestError{},
+			path:              "%zzzzz",
+		},
+	}
+
+	for _, test := range tests {
+		clientCfg := NewDefaultConfig()
+		config := configContainer{client: &clientCfg}
+		endpoint := newFakeMarathonEndpoint(t, &config)
+
+		client := endpoint.Client.(*marathonClient)
+
+		if test.clusterDown {
+			for _, member := range client.hosts.members {
+				member.status = memberStatusDown
+			}
+		}
+
+		_, _, err := client.buildAPIRequest("GET", test.path, nil)
+
+		if test.expectedError != nil {
+			assert.Equal(t, test.expectedError, err)
+		} else if test.expectedErrorType != nil {
+			assert.IsType(t, test.expectedErrorType, err)
+		} else {
+			assert.Fail(t, "Missing error assertion in test!")
```

Comment:
IMHO, this should be a panic since not providing either `expectedError` nor `expectedErrorType` is a programming mistake.

We might also want to move this check to the top prior to making the call to the system-under-test, similar to how input parameter validation happens first thing in a function call.

WDYT on both points?